addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -523,6 +516,18 @@ def _get_l10n_in_edi_ewaybill_line_details(self, line, line_tax_details, sign):
             line_details.update({"cessRate": self._l10n_in_round_value(tax_details_by_code.get("cess_rate"))})
         return line_details
 
+    def _get_l10n_in_edi_ewaybill_invalid_hsn_product_lines(self, move):
+        product_lines = []
+        for line in move.invoice_line_ids:
+            hsn_code = self._l10n_in_edi_extract_digits(line.l10n_in_hsn_code)
+            if (
+                line.display_type == 'product' and
+                not self._l10n_in_is_global_discount(line) and
+                not (hsn_code and re.match(r'^\d{4}$|^\d{6}$|^\d{8}$', hsn_code))
+            ):
+                product_lines.append(line)
```

Comment:
@jasa-odoo It would be better if you can remove this function and just use filtered instead of this.

Reply:
Hello @hamo-odoo ,
I already did that but @casp-odoo suggested this and IMO this is a better way because we need to write the same type of condition for an append error message(`account.edi.format`) and create an action(`account.move`) so instead of rewrite that we can manage with one function.